lib/spack/docs/basic_usage.rst

```diff
@@ -230,6 +230,54 @@ but you risk breaking other installed packages. In general, it is safer to
 remove dependent packages *before* removing their dependencies or use the
 ``--dependents`` option.
 
+
+.. _nondownloadable:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Non-Downloadable Tarballs
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The tarballs for some packages cannot be automatically downloaded by
+Spack.  This could be for a number of reasons:
+
+#. The author requires users to manually accept a license agreement
+   before downloading (``jdk`` and ``galahad``).
+
+#. The software is proprietary and cannot be downloaded on the open
+   Internet.
+
+To install these packages, one must create a mirror and manually add
+the tarballs in question to it (see :ref:`mirrors`):
+
+#. Create a directory for the mirror.  You can create this directory
+   anywhere you like, it does not have to be inside ``~/.spack``:
+
+   .. code-block:: console
+
+       $ mkdir ~/.spack/manual_mirror
+
+#. Register the mirror with Spack by creating ``~/.spack/mirrors.yaml``:
+
+   .. code-block:: yaml
+
+       mirrors:
+         manual: file:///home/me/.spack/manual_mirror
+
+#. Put your tarballs in it.  Tarballs should be named
+   ``<package>/<pacakge>-<version>.tar.gz``.  For example:
```

Comment:
The mirrors section has general instructions on how to  create a mirror.
This section shows how to create a MANUAL mirror, i.e. one that you don't
download everything into.  The steps are different.

On Wed, Sep 7, 2016 at 3:25 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In lib/spack/docs/basic_usage.rst:
>
> > +#. Create a directory for the mirror.  You can create this directory
> > +   anywhere you like, it does not have to be inside ``~/.spack``:
> > +
> > +   .. code-block:: console
> > +
> > +       $ mkdir ~/.spack/manual_mirror
> > +
> > +#. Register the mirror with Spack by creating ``~/.spack/mirrors.yaml``:
> > +
> > +   .. code-block:: yaml
> > +
> > +       mirrors:
> > +         manual: file:///home/me/.spack/manual_mirror
> > +
> > +#. Put your tarballs in it.  Tarballs should be named
> > +   ``<package>/<pacakge>-<version>.tar.gz``.  For example:
>
> To clarify, I agree that this section is useful to have here. But the
> instructions for how to create a mirror should probably be in Mirrors
> instead.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1676/files/c9e83c62a002c42852fcb54a3b45d54a68cb8ccc#r77887447>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cd6P1mQbNj5U8rNCean82BznhXVK_ks5qnw-NgaJpZM4JxOaa>
> .
>